src/_C014.ino

```diff
@@ -0,0 +1,686 @@
+#ifdef USES_C014
+//#######################################################################################################
+//################################# Controller Plugin 0014: Homie 3/4 ###################################
+//#######################################################################################################
+
+#define CPLUGIN_014
+#define CPLUGIN_ID_014              14
+
+// Define which Homie version to use
+//#define CPLUGIN_014_V3
+#define CPLUGIN_014_V4
+
+#ifdef CPLUGIN_014_V3
+  #define CPLUGIN_014_HOMIE_VERSION   "3.0.0"
+  #define CPLUGIN_NAME_014            "Homie MQTT (Version 3.0.1)"
+#endif
+#ifdef CPLUGIN_014_V4
+  #define CPLUGIN_014_HOMIE_VERSION   "4.0.0"
+  #define CPLUGIN_NAME_014            "Homie MQTT (Version 4.0.0 dev)"
+#endif
+
+// subscribe and publish shemes should not be changed by the user. This will probably break the homie convention. Do @ your own risk;)
+#define CPLUGIN_014_SUBSCRIBE       "homie/%sysname%/#" // "homie/%sysname%/+/+/#" causes problems (!ToDo)
+#define CPLUGIN_014_PUBLISH         "homie/%sysname%/%tskname%/%valname%"
+
+#define CPLUGIN_014_BASE_TOPIC      "homie/%sysname%/#"
+#define CPLUGIN_014_BASE_VALUE      "homie/%sysname%/%device%/%node%/%property%"
+#define CPLUGIN_014_INTERVAL        "90" // to prevent timeout !ToDo set by towest plugin interval
+#define CPLUGIN_014_SYSTEM_DEVICE   "SYSTEM" // name for system device Plugin for cmd and GIO values
+#define CPLUGIN_014_CMD_VALUE       "cmd" // name for command value
+#define CPLUGIN_014_GPIO_VALUE      "gpio" // name for gpio value i.e. "gpio1"
+#define CPLUGIN_014_CMD_VALUE_NAME  "Command" // human readabele name for command value
+
+byte msgCounter=0; // counter for sent Messages
+
+// !ToDo: should be integrated in bool CPlugin_014();
+bool CPlugin_014_Feedback(byte Function, struct EventStruct *event, String& str)
+{
+  MakeControllerSettings(ControllerSettings);
+  LoadControllerSettings(event->ControllerIndex, ControllerSettings);
+
+  if (str!="") {
+    String commandName = parseString(str, 1);
+    int errorCounter = 0;
+    if (commandName == F("gpio")) //!ToDo : As gpio is like any other plugin commands should be integrated below!
+    {
+      int port = parseString(str, 2).toInt();
+      int valueInt = parseString(str, 3).toInt();
+      String valueBool = "false";
+      if (valueInt==1) valueBool = "true";
+
+      String log = F("C014 : Feedback GPIO");
+      log += port;
+      log += F(" value:");
+      log += valueBool;
+      log += F(" (");
+      log += valueInt;
+      log += F(")");
+
+      String topic = ControllerSettings.Publish;
+      topic.replace(F("%sysname%"), Settings.Name);
+      topic.replace(F("%tskname%"), CPLUGIN_014_SYSTEM_DEVICE);
+      topic.replace(F("%valname%"), CPLUGIN_014_GPIO_VALUE + toString(port,0));
```

Comment:
Some of these template strings are also used in other string parse/replace functions.
Please check to see if they can be used or maybe conflict here.

Reply:
Changed the template to the CPLUGIN_014_PUBLISH constant. To stay in the homie standard the user should not be able to change this. So there will be no conflict, I think.